### DISCUSSION TITLE
Crowdsourced site targeting and coding

### DISCUSSION THREAD

**Discussion by WouldYouLikeToKnowMore** [W4775/Widify#5] [Ideas]
Request: a feature which enables an unlisted website to be cycled through various auto-attempts at widifying it, using stock code modifications (removal of various max-widths, etc), starting at a random point in the cyclic linked-list of options, and if one of them looks good, to 'lock' that in for the site. Toss in some back-end AI attempts if you want to get all 2022 about it. Let users go back to a previously generated set of lock information, so that if they're flicking idly through generated variants and don't realize they've run across a good one until after they clicked past it, they can return to it easily.

Second feature: those sitenames and choice of locks can be filtered back to a central repository or database. People can update or auto-update from that source. Sites in the db which have not been reviewed by the developer can then be cycled through the list of recorded lock options, in order of most common preference to least, by the users, followed by any non-tried options, and of course those users can then pick a lock option themselves to continue the preference accumulation.

Sites with strong user preferences for multiple types of lock options can then be manually implemented in the base code as having those options available, with the strongest option being the default. The manual nature of that step will help to weed out attempts at poisoning the code. The sites can be reviewed for manual inclusion in order of the number of locks submitted, so that the most popular sites for users of this extension get priority.

**Comment by W4775** [comment]
Hi! Thanks for the feedback. The first request is interesting as a lot of the process is currently manual. Each CSS file that WIDIFY uses is hand built by me to hopefully ensure that everything is working as intended. I did include a auto WIDIFY option in my last version but it was an all or nothing approach. Having the logic iterate through a site and strip away max-widths, etc could possibly work but would require a long set up process for the user. As I see it the logic would iterate through the page element by element looking for set max-widths. Then would strip out the max width and would allow the user to "lock" in the applied style, skip or cancel. Both lock and skip will move to the next element, so on and so forth. Might be an advanced user option.

 I have some ideas for a user curated site stylings to be added. I had started to get the ball rolling by extracting the embedded CSS files to a separate repo. I would love to be able to open that up to user created sheets or if the first option is implemented, the auto generated styles. I would perhaps put in a vote system that would allow me to keep track of which files are liked by users to eventually incorporate into base code.

Great ideas! I thank you greatly for trying out and/or using WIDIFY. I'm a single dev who works on this in his spare time, so your feedback and time are very much appreciated!!
